### ISSUE TITLE
Something strange happens to stdout encoding with `lein run`

### ISSUE THREAD

**Issue by jonathanj** [technomancy/leiningen#1857]
Running the following code via various means:

```clojure
(defn -main
  []
  (.write System/out (byte-array (map unchecked-byte
                                      [0xe2 0xe3 0xcf 0xd3]))))
```

```console
$ java -jar clojure-1.6.0.jar -e '(.write System/out (byte-array (map unchecked-byte [0xe2 0xe3 0xcf 0xd3])))' | xxd
0000000: e2e3 cfd3                                ....
```

```console
$ lein run | xxd
0000000: efbf bdef bfbd efbf bdef bfbd            ............
```

```console
lein trampoline run | xxd
0000000: e2e3 cfd3                                ....
````

`EFBFBD` is the UTF-8 encoding of `�` aka `REPLACEMENT CHARACTER`.

**Comment by jonathanj** [comment]
```console
$ lein version
Leiningen 2.5.1 on Java 1.7.0_45 Java HotSpot(TM) 64-Bit Server VM
```

On Mac OS X 10.10.2.

**Comment by hyPiRion** [comment]
Yeah, this is obviously a bug. I've found a (presumably working) solution, but it would require some changes to the test suite.

**Comment by pkalliok** [comment]
Which leiningen is supposed to have this fix?  I'm running leiningen 2.5.3 and I still see the same issue.
$ lein --version
Leiningen 2.5.3 on Java 1.8.0_72-internal OpenJDK 64-Bit Server VM
